pydantic/types.py

```diff
@@ -2344,9 +2344,11 @@ def __get_pydantic_json_schema__(
         return field_schema
 
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        base_schema: Any = handler(source)
+        del base_schema['type']
```

Comment:
Why is it necessary to remove the type from the base schema?

Reply:
Great question!